.github/workflows/go-tests.yml

```diff
@@ -376,7 +376,7 @@ jobs:
       runs-on: ${{ needs.setup.outputs.compute-xl }}
       repository-name: ${{ github.repository }}
       go-tags: "${{ github.event.repository.name == 'consul-enterprise' && 'consulent consulprem consuldev' || '' }}"
-      go-version: "1.19.10"
+      go-version: "1.19"
```

Comment:
I'm of two minds about this: on the one hand if we pin the patch version then we're on the hook to update our CI files every time Go releases a patch; on the other hand I agree with Dhia that if we don't pin it we could end up building/testing with different versions of Go in our pipelines. The latter is what happened between OSS and ENT. Our self hosted runners for the ENT repo picked up the new Go version and CI broke. The GH hosted runners were using the previous Go versions and this error didn't surface.

I think I prefer pinning only the `major.minor` version and letting CI pick the latest `patch` version to ease maintenance burden.

Reply:
question: do we need to make this change for the `1.20` flavours as well?

Reply:
I think all of our CI is set up to use `go.mod`.  This is the only case where we are testing on a previous version.  I think this might warrant a discussion outside of this PR because if we change it, we will want to make this across "everything" or at least much more than this.  I think that would be worth getting consensus on and also documenting in the PR description (now the commit message in main~).  I'll start a thread.

Reply:
@cthain thank you for calling that out.  this is fixed in f8578b0749fe3df10906b53f91416e9f2421bc29